pkg/sessions/redis/redis_store.go

```diff
@@ -114,20 +126,33 @@ func (store *SessionStore) SaveUserSession(ctx context.Context, s *sessions.Sess
 	} else {
 		ticket = string(keyVal)
 	}
-	err := s.ObtainLock(ctx, exp)
-	if err != nil {
-		return fmt.Errorf("error occurred while trying to obtain lock: %v", err)
-	}
-	defer func() {
-		if s == nil {
-			return
+	return saveTicketWithLock(ctx, store, keyName, ticket, exp)
+}
+
+func (store *SessionStore) ClearUserSession(ctx context.Context, s *sessions.SessionState, value string, exp time.Duration) error {
```

Comment:
Similar comment about the locking in this clear function... the lock needs to be obtained before the "read" in the "read-modify-write".

Reply:
done @joel-bluedata . thanks